lib/gem_publisher/git_remote.rb

```diff
@@ -18,6 +18,12 @@ def add_tag(tag_name, commit_ish = "HEAD")
       git "push", @remote_name, "tag", tag_name
     end
 
+    def add_branch(branch_name, commit_ish = "HEAD")
+      sha1 = git("rev-parse", commit_ish).chomp
+      git "push", @remote_name, "#{sha1}:#{branch_name}"
```

Comment:
I'm not sure this git command is quite correct. It'll fail if the remote branch doesn't already exist.  I think if you change it to `git "push", @remote_name, "#{sha1}:refs/heads/#{branch_name}"` It will create the branch if it doesn't already exist.